omod/src/main/java/org/openmrs/module/fhir2/web/filter/SmartAuthenticationFilter.java

```diff
@@ -65,29 +67,56 @@ public void init(FilterConfig filterConfig) {
 	public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain) throws IOException, ServletException {
 		if (req instanceof HttpServletRequest) {
 			final HttpServletRequest httpRequest = (HttpServletRequest) req;
-			
-			// KeycloakOIDCFilter does the actual request handling
-			super.doFilter(req, res, (rq, rs) -> {});
-			
 			if (httpRequest.getRequestedSessionId() != null && !httpRequest.isRequestedSessionIdValid()) {
 				Context.logout();
 			}
 			
-			if (httpRequest.getAttribute(KeycloakAccount.class.getName()) != null) {
-				KeycloakAccount account = (KeycloakAccount) httpRequest.getAttribute(KeycloakAccount.class.getName());
+			if (!Context.isAuthenticated()) {
+				// KeycloakOIDCFilter does the actual request handling
+				super.doFilter(req, res, (rq, rs) -> {});
 				
-				User user = new User();
-				user.setUsername(account.getPrincipal().getName());
-				
-				Authenticated authenticated = Context.authenticate(new SmartTokenCredentials(user));
+				if (httpRequest.getRequestedSessionId() != null && !httpRequest.isRequestedSessionIdValid()) {
+					Context.logout();
+				}
 				
-				log.debug("The user '{}' was successfully authenticated as OpenMRS user {}",
-				    account.getPrincipal().getName(), authenticated.getUser());
-			} else {
-				if (!res.isCommitted()) {
-					HttpServletResponse httpResponse = (HttpServletResponse) res;
-					httpResponse.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Not authenticated");
-					return;
+				if (httpRequest.getAttribute(KeycloakAccount.class.getName()) != null) {
+					OidcKeycloakAccount account = (OidcKeycloakAccount) httpRequest
+					        .getAttribute(KeycloakAccount.class.getName());
+					
+					String userName = account.getKeycloakSecurityContext().getToken().getSubject();
```

Comment:
@ibacher This still give the internal keycloak user id

Reply:
Ok, use that.